src/TurnerSoftware.SitemapTools/SitemapQuery.cs

```diff
@@ -82,6 +83,8 @@ public async Task<IEnumerable<Uri>> DiscoverSitemapsAsync(string domainName)
 			};
 			
 			var robotsFile = await new RobotsFileParser(HttpClient).FromUriAsync(baseUri);
```

Comment:
What I might do is add `CancellationToken` support [in my robots parsing library](https://github.com/TurnerSoftware/RobotsExclusionTools) so we can pass your newly added cancellation token there too.

Reply:
It will good. But while I stay it here.